azurerm/resource_arm_mariadb_server.go

```diff
@@ -0,0 +1,381 @@
+	d.Set("ssl_enforcement", string(resp.SslEnforcement))
+
+	if err := d.Set("sku", flattenMariaDbServerSku(resp.Sku)); err != nil {
+		return fmt.Errorf("Error setting `sku`: %+v", err)
+	}
+
+	if err := d.Set("storage_profile", flattenMariaDbStorageProfile(resp.StorageProfile)); err != nil {
+		return fmt.Errorf("Error setting `storage_profile`: %+v", err)
+	}
+
+	flattenAndSetTags(d, resp.Tags)
+
+	// Computed
+	d.Set("fqdn", resp.FullyQualifiedDomainName)
+
+	return nil
+}
+
+func resourceArmMariaDbServerDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).mariadbServersClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["servers"]
+
+	future, err := client.Delete(ctx, resourceGroup, name)
+	if err != nil {
+		if response.WasNotFound(future.Response()) {
+			return nil
+		}
+
+		return fmt.Errorf("Error deleting MariaDB Server %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	err = future.WaitForCompletionRef(ctx, client.Client)
+	if err != nil {
+		if response.WasNotFound(future.Response()) {
+			return nil
+		}
+
+		return fmt.Errorf("Error waiting for deletion of MariaDB Server %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	return nil
+}
+
+func expandAzureRmMariaDbServerSku(d *schema.ResourceData) *mariadb.Sku {
+	skus := d.Get("sku").([]interface{})
+	sku := skus[0].(map[string]interface{})
+
+	name := sku["name"].(string)
+	capacity := sku["capacity"].(int)
+	tier := sku["tier"].(string)
+	family := "Gen5"
+
+	return &mariadb.Sku{
+		Name:     utils.String(name),
+		Tier:     mariadb.SkuTier(tier),
+		Capacity: utils.Int32(int32(capacity)),
+		Family:   utils.String(family),
+	}
+}
+
+func expandAzureRmMariaDbStorageProfile(d *schema.ResourceData) *mariadb.StorageProfile {
+	storageprofiles := d.Get("storage_profile").([]interface{})
+	storageprofile := storageprofiles[0].(map[string]interface{})
+
+	backupRetentionDays := storageprofile["backup_retention_days"].(int)
+	geoRedundantBackup := storageprofile["geo_redundant_backup"].(string)
+	storageMB := storageprofile["storage_mb"].(int)
+
+	return &mariadb.StorageProfile{
+		BackupRetentionDays: utils.Int32(int32(backupRetentionDays)),
+		GeoRedundantBackup:  mariadb.GeoRedundantBackup(geoRedundantBackup),
+		StorageMB:           utils.Int32(int32(storageMB)),
+	}
+}
+
+func flattenMariaDbServerSku(resp *mariadb.Sku) []interface{} {
+	values := map[string]interface{}{}
+
+	if name := resp.Name; name != nil {
+		values["name"] = *name
+	}
+
+	if capacity := resp.Capacity; capacity != nil {
+		values["capacity"] = *capacity
+	}
+
+	values["tier"] = string(resp.Tier)
+
+	if family := resp.Family; family != nil {
+		values["family"] = *family
+	}
+
+	return []interface{}{values}
+}
+
+func flattenMariaDbStorageProfile(resp *mariadb.StorageProfile) []interface{} {
```

Comment:
(same here) can we return an empty `[]interface{}{}` when `resp` is nil?

Reply:
Sure.